CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift

```diff
@@ -548,39 +548,6 @@ public let DECL_NODES: [Node] = [
     elementChoices: [.designatedType]
   ),
 
-  Node(
-    kind: .editorPlaceholderDecl,
-    base: .decl,
-    nameForDiagnostics: "editor placeholder",
```

Comment:
@ahoppen we don't have nay functionally for setting an complete node as deprecated.
Should we add it?

Reply:
Sorry for the delayed response. We needed to discuss how to handle this. 

We should deprecate the nodes. The parser should just stop generating them but that way they will still exist for anyone using swift-syntax for code generation and it will make transitioning to a new swift-syntax version easier because the deprecation warnings can guide you in the right direction.

Reply:
Would the best approach to to just copy the needed code and place it in the deprecated files or should we introduce a new field on the `Node` where we can deprecate it?

Because we also need to ensure the parser have the `open func visit(_ node: EditorPlaceholderDeclSyntax) -> DeclSyntax` that is deprecated. 
Otherwise it will break

Reply:
I think we need to properly generate all the generated cases/functions using CodeGeneration. Maintaining them manually won’t scale, I think.

Reply:
As I tought. Just needed to confirm